### PR TITLE
CMakeLists.txt: Clang-warnings: removed -Wno-sign-conversion

### PR DESCRIPTION
**Pull request by medithe**
I agree that my contributions are licensed under the {fmt} license, and agree to future changes to the licensing.

### CODE REVIEW

File: test/format-test.cc [hunk 1]
```diff
@@ -1828,14 +1828,14 @@ class mock_arg_formatter
   }
 
   template <typename T>
-  typename std::enable_if<fmt::detail::is_integral<T>::value, iterator>::type
+  typename std::enable_if<fmt::detail::is_integral<T>::value && fmt::detail::is_signed<T>::value, iterator>::type
```
Comment (vitaut) [inline test/format-test.cc:1831]: Please apply clang-format.

Comment (medithe) [reply test/format-test.cc]: Applied

**Comment by vitaut** [comment]
Thanks!

File: CMakeLists.txt [hunk 2]
```diff
@@ -130,7 +130,7 @@ endif ()
 
 if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   set(PEDANTIC_COMPILE_FLAGS -Wall -Wextra -pedantic -Wconversion
-      -Wno-sign-conversion -Wdeprecated -Wweak-vtables)
```
Comment (0x8000-0000) [inline CMakeLists.txt:133]: Why was this removed?
